File: frontend/types/rag.ts
export type RAGSource = {
  memory_id: number;
  title: string;
  category: string;
  distance: number;
};

export type RAGRequest = {
  question: string;
  top_k?: number;
};

export type RAGResponse = {
  answer: string;
  sources: RAGSource[];
};

export type ChatMessage = {
  id: string;
  role: "user" | "assistant";
  content: string;
  sources?: RAGSource[];
};

File: backend/app/schemas/rag.py
from pydantic import BaseModel, Field


class RAGRequest(BaseModel):
    question: str

    top_k: int = Field(
        default=5,
        ge=1,
        le=20
    )


class RAGSourceMemory(BaseModel):
    memory_id: int
    title: str
    category: str
    distance: float


class RAGResponse(BaseModel):
    answer: str

    sources: list[RAGSourceMemory]